arelle/plugin/validate/EDINET/rules/gfm.py

```diff
@@ -1256,6 +1256,39 @@ def rule_gfm_1_3_31(
         )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_gfm_1_4_4(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC5700W: [GFM 1.4.4] If the type attribute is "extended" or "resource", set the xlink:role attribute to the extended link role.
+
+    GFM 1.4.4: The xlink:role attribute of an element with a type="extended" attribute or a
+    type="resource" attribute must be present and must not be empty.
+    """
+
+    for modelDocument in val.modelXbrl.urlDocs.values():
+        if pluginData.isStandardTaxonomyUrl(modelDocument.uri, val.modelXbrl):
+            continue
+        rootElt = modelDocument.xmlRootElement
+        for elt in rootElt.iterdescendants():
```

Comment:
To avoid a full iteration over all elements, you could do something like:
```
ns = {'xlink': 'http://www.w3.org/1999/xlink'}
for elt in rootElt.xpath('//*[@xlink:type="extended" or @xlink:type="resource"]', namespaces=ns):
```